proposed/template-renderer.md

```diff
@@ -0,0 +1,209 @@
+
+The word `implementor` in this document is to be interpreted as someone
+implementing the `TemplateRendererInterface` in a [template engine][] library.
+Users of template renderer are referred to as `user`.
+The word `enduser` in this document is to be interpreted as someone using
+a library, framework, cms created by a `user`.
+
+[RFC 2119]: http://tools.ietf.org/html/rfc2119
+
+## Goal
+
+Having common interfaces for rendering templates allows developers to create libraries that can interact with many frameworks and other libraries in a common fashion.
+
+Some examples may use the Interface:
+
+ - Content Management Systems (Sulu CMS, Drupal, Typo3, Contao CMS, ...)
+ - E-Commerce Platforms (Sylius, Spryker)
+ - Newsletter/Mail Libraries (Symfony Mailer, Abstractions over Mailchimp and other Newsletter tools) 
+ - Anything following ["Separating content from presentation"](https://en.wikipedia.org/wiki/Separation_of_content_and_presentation) where the presentation is project specific.
+
+Some examples may implement the Interface or providing a bridge:
+
+ - Blade (Laravel Framework)
+ - Laminas View
+ - Latte (Nette Framework)
+ - Smarty
+ - Sulu CMS
+ - Twig (Symfony Framework)
+ - Typo3
+
+## Definitions
+
+* **Template** - A string representation of a template.
+* **Context** - A array of context data given into to the rendered template.
+* **TemplateRenderer** - A service rendering which will render the template with the given context and return the rendered content.
+
+### Template
+
+A template MUST be a string representation of a given template supported by the template renderer. It MAY be a file path
+to the template file, but it can also be a virtual name or path supported only by a specific template renderer. The
+template is not limited by specific characters by definition but a template renderer MAY support only specific one.
+
+### Context
+
+A context is MUST be an array of the available variables given to the template renderer. The array keys represent the
+available variables and MUST be typed as string. The array values represent the variables value and can be anything and
+are so typed by mixed. The context SHOULD NOT be an object to support also basic implementation of a template renderer.
+
+### TemplateRenderer
+
+A template renderer is a service object implementing the `TemplateRendererInterface`. It MUST be responsible to render a 
+supported template by a OPTIONAL context. It MUST return the rendered content of the template as a string.
+If a template was not found by the template renderer, an Exception implementing the `TemplateNotFoundExceptionInterface`
+MUST be thrown. The template renderer SHALL NOT in any case output/print/stream something directly to php output.
+A implementor of a template renderer is allowed to support as context also an object, but a user is REQUIRED whennusing
+The `TemplateRendererInterface` to give the context as array to support also simple template renderers.
```

Comment:
I agree and maybe it did just confuse here and better to remove that sentence about object here. The context will always be a array in the Interface, just the contravariance support of PHP allow that it could also be `array|object` that is something we can not even prevent we just can recommend that it should be always an array. Really thx  for your feedback  and insights about how yii views work, I will have a look at the other comments later.